test/kes_dashboard_live_tiles_cli/TestOnlineLoader.h

```diff
@@ -12,24 +12,265 @@
+
+    // Set the HTTPClient.GET to return a given valid KES response.
+    ON_CALL(*mockHttpClient, GET_impl(OnlineLoader::KES_DLT_URL, _))
+        .WillByDefault(Return(this->response));
+
+    // Return the same mock Tile each for each tile data from the response.
+    ON_CALL(*mockTileFactory, create)
+        .WillByDefault(Return(mockTile));
+
+    // Ignore the call to Tile.initialise().
+    ON_CALL(*mockTile, initialise)
+        .WillByDefault(Return(true));
+
+    // Ignore the call to Tile.download().
+    ON_CALL(*mockTile, download)
+        .WillByDefault(Return(true));
+
+    EXPECT_CALL(*mockTileFactory, create)
+        .Times(this->tileCount);
+
+    onlineLoader.getTiles();
+}
+
+
+MATCHER_P(AnyInJsonArray, jsonArray, "") {
+    for (int i = 0; i < json_array_get_count(jsonArray); i++)
+        if (json_value_equals(json_array_get_value(jsonArray, i), arg))
+            return true;
+    return false;
+}
+
+
+/**
+ * Check that for a given HTTPClient.GET() response OnlineLoader.getTiles()
+ * calls for each tile data Tile.initialise() with the expected JSON.
+ */
+TEST_P(KesDltValidResponsesFixture, GetTilesCallsTileInitialise) {
+    shared_ptr<MockHTTPClient> mockHttpClient = make_shared<MockHTTPClient>();
+    shared_ptr<MockTileFactory> mockTileFactory = make_shared<MockTileFactory>();  // NOLINT
+    shared_ptr<MockTile> mockTile = make_shared<MockTile>();
+
+    OnlineLoader onlineLoader(
+        "/tmp/kes-dlt-cli/test", mockHttpClient, mockTileFactory);
+
+    // Set the HTTPClient.GET to return a given valid KES response.
+    ON_CALL(*mockHttpClient, GET_impl(OnlineLoader::KES_DLT_URL, _))
+        .WillByDefault(Return(this->response));
+
+    // Return the same mock Tile each for each tile data from the response.
+    ON_CALL(*mockTileFactory, create)
+        .WillByDefault(Return(mockTile));
+
+    // Ignore the call to Tile.initialise().
+    ON_CALL(*mockTile, initialise)
+        .WillByDefault(Return(true));
+
+    // Ignore the call to Tile.download().
+    ON_CALL(*mockTile, download)
+        .WillByDefault(Return(true));
+
+    JSON_Array* tilesData = json_object_get_array(
+        json_value_get_object(this->response.get()), "shares");
+
+    // TODO: Should use AllInJsonArray to check that each item passed to
+    // initialise is at a unique index in a given array. No clue how to atm.
+    EXPECT_CALL(*mockTile, initialise(AnyInJsonArray(tilesData)))
+        .Times(this->tileCount);
+
+    onlineLoader.getTiles();
+}
+
+/**
+ * Check that for a given HTTPClient.GET() response OnlineLoader.getTiles()
+ * calls for each tile data Tile.download() with the given cacheDir.
+ */
+TEST_P(KesDltValidResponsesFixture, GetTilesCallsTileDownload) {
+    shared_ptr<MockHTTPClient> mockHttpClient = make_shared<MockHTTPClient>();
+    shared_ptr<MockTileFactory> mockTileFactory = make_shared<MockTileFactory>();  // NOLINT
+    shared_ptr<MockTile> mockTile = make_shared<MockTile>();
```

Comment:
I think that this is one of the few cases where you might actually want to use `auto`.